kata_proxy.go

```diff
@@ -84,15 +96,56 @@ func (p *kataProxy) unregister(pod Pod) error {
 
 // connect is kataProxy connect implementation for proxy interface.
 func (p *kataProxy) connect(pod Pod, createToken bool) (ProxyInfo, string, error) {
+	client, err := kataclient.NewAgentClient(pod.state.URL)
+	if err != nil {
+		return ProxyInfo{}, "", err
+	}
+
+	p.client = client
+
 	return ProxyInfo{}, p.proxyURL, nil
 }
 
 // disconnect is kataProxy disconnect implementation for proxy interface.
 func (p *kataProxy) disconnect() error {
+	if p.client == nil {
+		return fmt.Errorf("Client is nil, we can't interact with kata-proxy")
+	}
+
+	p.client.Close()
+
 	return nil
 }
 
 // sendCmd is kataProxy sendCmd implementation for proxy interface.
 func (p *kataProxy) sendCmd(cmd interface{}) (interface{}, error) {
-	return nil, nil
+	if p.client == nil {
+		return nil, fmt.Errorf("Client is nil, we can't interact with kata-proxy")
+	}
+
+	switch c := cmd.(type) {
```

Comment:
I don't like this whole commit `kata_proxy: Implement all interface ops` because we are removing the support for `no_proxy` in that case... Indeed, the goal of no proxy is to bypass the proxy. I understand what we are trying to do here but I am wondering if we shouldn't avoid this kind of patch.